src/SalesShareDistributor.sol

```diff
@@ -85,19 +91,20 @@ contract SalesShareDistributor is UUPSUpgradeable, OwnableUpgradeable, Reentranc
 
     /**
      * @notice anyone should be able to call this function after having observed the sale
-     *         rn we restrict it to the token issuer since they must provide a permissioner that controls the claiming rules
+     *         rn we restrict it to the original owner since they must provide a permissioner that controls the claiming rules
      *         this is a deep dependency on our own sales contract
      *
      * @param tokenContract IPToken     the tokenContract of the IPToken
      * @param listingId     uint256     the listing id on Schmackoswap
      * @param permissioner  IPermissioner   the permissioner that permits claims
      */
     function afterSale(IPToken tokenContract, uint256 listingId, IPermissioner permissioner) external {
-        if (_msgSender() != tokenContract.issuer()) {
+        Metadata memory metadata = tokenContract.metadata();
+        //todo: this should be the *former* holder of the IPNFT, not the 1st owner :)
```

Comment:
we mean here the current owner of the IPNFT?

Reply:
this contract is not used. When there are todos in a contract codebase we should consider it not production ready. The idea of this concept is that the *previous* owner of an IPNFT receives sales proceedings and distributes them among the former holders.